rcgen/src/lib.rs

```diff
@@ -611,6 +611,15 @@ impl CertificateParams {
 		let name_constraints = Self::convert_x509_name_constraints(&x509)?;
 		let serial_number = Some(x509.serial.to_bytes_be().into());
 
+		let mut key_identifier_method = KeyIdMethod::Sha256;
+		for ext in x509.extensions() {
+			if let x509_parser::extensions::ParsedExtension::SubjectKeyIdentifier(v) =
+				ext.parsed_extension()
+			{
+				key_identifier_method = KeyIdMethod::PreSpecified(v.0.to_vec());
+			}
+		}
```

Comment:
I think this might be a little bit tidier using `iter_extensions().find_map()`, we can drop the `mut` this way:

```rust
		let key_identifier_method = x509
			.iter_extensions()
			.find_map(|ext| match ext.parsed_extension() {
				x509_parser::extensions::ParsedExtension::SubjectKeyIdentifier(key_id) => {
					Some(KeyIdMethod::PreSpecified(key_id.0.into()))
				},
				_ => None,
			})
			.unwrap_or(KeyIdMethod::Sha256);
```

Reply:
@cpu updated as requested